test/suite/test_cursor17.py

```diff
@@ -95,12 +95,12 @@ def test_cursor_skip_pages(self):
         # Check if we skipped any pages while moving the cursor.
         #
         # We calculate the number of pages we expect to skip based on the total number of leaf pages
-        # reported in the WT stats. We subtract 2 from the count of leaf pages in the table and test
-        # that we atleast skipped 80% of the expected number of pages.
-
+        # reported in the WT stats. We also skip internal pages, hence skipping the parts of the
+        # whole tree wherever we can. The number of pages skipped depends a lot on the tree
+        # structure. It is a good test to check if we skipped atleast 50% of the leaf pages.
```

Comment:
This test fails now because we are not visiting each leaf page and deciding to skip or not. We are skipping at a higher level, which means we are making less decisions on how many pages we skip, but lead to walking less leaf pages eventually. The test is written as a percentage of leaf pages seen vs skipped. @hbokhari , can you think of an alternative test scenario ?

Reply:
Discussed with @hbokhari and added a test to check that we skipped at least 1 page.